service/dynamodb/auth.go

```diff
@@ -15,6 +16,42 @@ func bindAuthParamsRegion(params *AuthResolverParameters, _ interface{}, options
 	params.Region = options.Region
 }
 
+type setLegacyContextSigningOptionsMiddleware struct {
+}
+
+func (*setLegacyContextSigningOptionsMiddleware) ID() string {
+	return "setLegacyContextSigningOptions"
+}
+
+func (m *setLegacyContextSigningOptionsMiddleware) HandleFinalize(ctx context.Context, in middleware.FinalizeInput, next middleware.FinalizeHandler) (
+	out middleware.FinalizeOutput, metadata middleware.Metadata, err error,
+) {
+	rscheme := getResolvedAuthScheme(ctx)
+	schemeID := rscheme.Scheme.SchemeID()
+
+	if sn := awsmiddleware.GetSigningName(ctx); sn != "" {
```

Comment:
do we know of an actual occurrence where users were setting the signing name manually on the context? or is it just our code and the possibility of users being able to do this that we need to support it?

Reply:
Someone from lightsail was definitely doing it after the endpoints2 refactor (v1 resolution let you bake in a signing region, v2 doesn't).